.eslintrc.js

```diff
@@ -1,63 +1,242 @@
+          objectLiteralTypeAssertions: 'allow-as-parameter',
+        }],
+
+        // Ensure that the values returned from a module are of the expected type
+        '@typescript-eslint/explicit-module-boundary-types': ['error', {
+          allowArgumentsExplicitlyTypedAsAny: true,
+        }],
+
+        // Turns off all JSDoc plugin rules because they don't work consistently in TypeScript contexts. For example,
+        // it's not an error to export interfaces and types that don't have JSDoc on them without these contexts. Also,
+        // satisfying some of these rules would require redundant type information in the JSDoc comments, so its in
+        // conflict with the next rule.
+        // TODO: track progress on this issue https://github.com/gajus/eslint-plugin-jsdoc/issues/615
+        ...jsDocRecommendedRulesOff,
+
+        // No types in JSDoc for @param or @returns. TypeScript will provide this type information, so it would be
+        // redundant, and possibly conflicting.
+        'jsdoc/no-types': 'error',
+
+        // Allow use of import and export syntax, despite it not being supported in the node versions. Since this
+        // project is transpiled, the ignore option is used. Overrides node/recommended.
+        // 'node/no-unsupported-features/es-syntax': ['error', { ignores: ['modules'] }],
+        // TODO: The node plugin's ignore option doesn't work in order to suppress this error.
+        'node/no-unsupported-features/es-syntax': 'off',
+
+        // Allow safe references to functions before the declaration. Overrides AirBnB config. Not located in the
+        // override section below because a distinct override is necessary in JavaScript files.
+        'no-use-before-define': 'off',
+        '@typescript-eslint/no-use-before-define': ['error', 'nofunc'],
+      },
+    },
+    {
+      files: ['**/*.js', '**/*.ts'],
+      rules: {
+        // Override rules
+        // ---
+        // This level of this configuration contains rules which override options or disable rules in the base
+        // configurations in both JavaScript and TypeScript.
+
+        // Increase the max line length to 120. The rest of this setting is copied from the AirBnB config.
+        'max-len': ['error', 120, 2, {
+          ignoreUrls: true,
+          ignoreComments: false,
+          ignoreRegExpLiterals: true,
+          ignoreStrings: true,
+          ignoreTemplateLiterals: true,
+        }],
+
+        // Restrict the use of backticks to declare a normal string. Template literals should only be used when the
+        // template string contains placeholders. The rest of this setting is copied from the AirBnb config.
+        quotes: ['error', 'single', { avoidEscape: true, allowTemplateLiterals: false }],
+
+        // TODO: not sure if this will work because of plugin loading on .js files
+        // Allow leading underscores for parameter names, which is used to acknowledge unused variables in TypeScript.
```

Comment:
This is the only rule that gives me some semblance of a pause (more for the enforced camelCase rather than leading underscores), but I'm OK with incorporating it and seeing how it goes!

Reply:
Is there a particular other variable naming scheme you'd prefer to use? Or would you rather have complete freedom in variable naming (that is, no enforcement of variable naming)? Related to this topic is the first mini-PR I created for this PR in #1025, where type properties currently used in various `src/types` files sometimes use camelCase and sometimes use snake_case.

Reply:
Complete freedom with strong guidance/agreement amongst maintainers -- reluctantly and only because I think the linter doesn't allow the happy medium outlined below.

I think it should almost always be camelCase, but prefer not to be bound to that with zero exceptions made. For example, what comes to mind that caused the brief pause was if there is a magic number of some sort that I want to use a global case/formatting for. Or in our example docs, the used variable name `myDB` comes to mind (though maybe that would be permissible? 🤔 ). Small stuff like that!

I definitely think the instances where it's been used outside of the cases above (like in the linked PR) should be addressed at some point and any regression avoided after the fixes are introduced. I'm very much for adhered-to standards and patterns in those cases, though that code also pre-dates me!

Reply:
Sounds good and makes a ton of sense. What do you think of downgrading the camelCase naming convention for variables from an error to warning? This way there is still some kind of noise made by the tooling when you don't use camelCase, but not enough for it to get in the way by failing the tests?

I'll look into the configurability of this rule as I think we can have it apply rules quite contextually; I'll point once more to #1025 as an example of applying different rules to type properties specifically, maybe there are other kinds of variables we can target in a similar manner to try to get closer to enforcing the sort of style you are looking for.

Reply:
Perf, that works! Though it brings up the question as to how tolerant we intend to be of warnings and if warnings do pop up, how do we ensure they're caught and addressed if not considered passable exceptions? But again, we can cross that bridge when we get there. These changes are a massive improvement, as is, and it was just the smallest of pauses! And if we just want to revisit this at a later date, also totally fine with that. 🙂